# 1. How many observations are in the data?

In [ ]:
import numpy as np
import pandas as pd
import sklearn

In [ ]:
data = pd.read_csv('/content/column_2C.csv')

In [ ]:
data.head
# in target class: 1 = abnormal, 0 = normal

<bound method NDFrame.head of      pelvic_incidence  pelvic_tilt  lumbar_lordosis_angle  sacral_slope  \
0           63.027817    22.552586              39.609117     40.475232   
1           39.056951    10.060991              25.015378     28.995960   
2           68.832021    22.218482              50.092194     46.613539   
3           69.297008    24.652878              44.311238     44.644130   
4           49.712859     9.652075              28.317406     40.060784   
..                ...          ...                    ...           ...   
305         47.903565    13.616688              36.000000     34.286877   
306         53.936748    20.721496              29.220534     33.215251   
307         61.446597    22.694968              46.170347     38.751628   
308         45.252792     8.693157              41.583126     36.559635   
309         33.841641     5.073991              36.641233     28.767649   

     pelvic_radius  degree_spondylolisthesis     class  
0        98.

In [ ]:
data.shape

(310, 7)

*There are 310 observations in the data for 7 variables with the class of orthopedic patients*

# 2 Build a logistic regression model. Examine whether the scaling improves the fit of the model. You can choose any scaling method for the examination. 

In [ ]:
# divide data into x and y
# split data into training and test (30%)
X=data.drop(['class'], axis='columns')
y=data['class']

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.3,random_state = 0)

In [ ]:
# build model and train the model with training set
from sklearn.linear_model import LogisticRegression
logis = LogisticRegression()
logis.fit(X_train, y_train)

LogisticRegression()

In [ ]:
# Coefficients
df=pd.DataFrame(logis.coef_.T, X.columns,columns=['Coefficient'])
print(df)

                          Coefficient
pelvic_incidence             0.003495
pelvic_tilt                 -0.077930
lumbar_lordosis_angle        0.042184
sacral_slope                 0.081395
pelvic_radius                0.120309
degree_spondylolisthesis    -0.198046


In [ ]:
# accuracy
y_pred=logis.predict(X_test)
from sklearn import metrics
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.7849462365591398


In [ ]:
# Standard Scaler
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()

scaler.fit(X_train)
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [ ]:
# Linear regression with scale 
logis_scale = LogisticRegression() 
logis_scale.fit(X_train_scaled,y_train)

LogisticRegression()

In [ ]:
# accuracy
y_pred=logis_scale.predict(X_test_scaled)
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.8172043010752689


*Here, i have built a Logistic Regression model with 78% accuracy. And, based on the examination of feature scaling, it increases the fit of the model to 82% accuracy by 4 percentages, and i have used the Standard Scaler method for the examination.*

# 3 Build a K-Nearest Neighbor model. Examine whether the scaling improves the fit of the model. You can choose any scaling method for the examination. 

In [101]:
# build model and train the model with training set
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=7)
knn.fit(X_train, y_train)

KNeighborsClassifier(n_neighbors=7)

In [102]:
# accuracy
y_pred=knn.predict(X_test)
from sklearn import metrics
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.8172043010752689


In [103]:
# Standard Scaler
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()

scaler.fit(X_train)
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [104]:
# K-Nearest Neighbors Classifier with scale 
knn_scale = KNeighborsClassifier() 
knn_scale.fit(X_train_scaled,y_train)

KNeighborsClassifier()

In [105]:
# accuracy
y_pred=knn_scale.predict(X_test_scaled)
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.7849462365591398


*Here, i have built a K-Nearest Neighbor model with 82% accuracy. And, based on the examination of feature scaling, it decreases the fit of the model to 78% accuracy by 4 percentages, and i have used the Standard Scaler method for the examination.*

# 4. Build a support vector machine model. Examine whether the scaling improves the fit of the model. You can choose any scaling method for the examination

In [75]:
# build model and train the model with training set
from sklearn.svm import SVC
svm = SVC(kernel='linear',random_state=42)
svm.fit(X_train, y_train)

SVC(kernel='linear', random_state=42)

In [76]:
# accuracy
y_pred=svm.predict(X_test)
from sklearn import metrics
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.8172043010752689


In [81]:
# Standard Scaler
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()

scaler.fit(X_train)
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [82]:
# Support Vector Machine with scale 
svm_scale = SVC() 
svm_scale.fit(X_train_scaled,y_train)

SVC()

In [83]:
# accuracy
y_pred=svm_scale.predict(X_test_scaled)
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.8064516129032258


*Here, i have built a Support Vector Machine model with 82% accuracy. And, based on the examination of feature scaling, it decreases the fit of the model to 81% by  percentage, and i have used the Standard Scaler method for the examination.*

# 5. Build a decision tree model. Examine whether the scaling improves the fit of the model.You can choose any scaling method for the examination

In [85]:
# build model and train the model with training set
from sklearn.tree import DecisionTreeClassifier
dtree = DecisionTreeClassifier(max_depth=4,random_state=0)
dtree.fit(X_train, y_train)

DecisionTreeClassifier(max_depth=4, random_state=0)

In [86]:
# accuracy
y_pred=dtree.predict(X_test)
from sklearn import metrics
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.8064516129032258


In [87]:
# Standard Scaler
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()

scaler.fit(X_train)
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [88]:
# Decision Tree with scale 
dtree_scale = DecisionTreeClassifier() 
dtree_scale.fit(X_train_scaled,y_train)

DecisionTreeClassifier()

In [89]:
# accuracy
y_pred=dtree_scale.predict(X_test_scaled)
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.8172043010752689


*Here, i have built a Decision Tree model with 81% accuracy. And, based on the examination of feature scaling, it increases the fit of the model to 82% by 1 percentage, and i have used the Standard Scaler method for the examination.*

# 6. Build a random forest model. Examine whether the scaling improves the fit of the model.You can choose any scaling method for the examination. 

In [91]:
# build model and train the model with training set
from sklearn.ensemble import RandomForestClassifier
rforest = RandomForestClassifier(n_estimators=10,random_state=0)
rforest.fit(X_train, y_train)

RandomForestClassifier(n_estimators=10, random_state=0)

In [92]:
# accuracy
y_pred=rforest.predict(X_test)
from sklearn import metrics
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.7849462365591398


In [93]:
# Standard Scaler
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()

scaler.fit(X_train)
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [94]:
# Random Forest with scale 
rforest_scale = RandomForestClassifier() 
rforest_scale.fit(X_train_scaled,y_train)

RandomForestClassifier()

In [95]:
# accuracy
y_pred=rforest_scale.predict(X_test_scaled)
print("Accuracy",metrics.accuracy_score(y_test,y_pred))

Accuracy 0.8064516129032258


*Here, i have built a Random Forest model with 78% accuracy. And, based on the examination of feature scaling, it increases the fit of the model to 81% by 3 percentages, and i have used the Standard Scaler method for the examination.*

# 7. Among the five model you examined, which one do you think is the best fit for the data?

*Among the five models from the examination, i have observed that the best fit model for the data are equally K-Nearest Neighbor and Support Vector Machine with small percent differences compared to others models without scalling. 
With Scaling, the best fit for the data is Logistic Regression and Decision Tree. One more observations all models generally works pretty well on this data without much differences in between.*